components/HomepageSection/PowersAIAgentEconomies.jsx

```diff
@@ -0,0 +1,578 @@
+          icon="agent-to-agent.png"
+          iconWidth={104}
+          iconHeight={36}
+          primaryValue={processedMetrics?.a2aTransactions}
+          primaryText="A2A txns"
+          primaryLink={DUNE_A2A_TRANSACTIONS_QUERY_URL}
+          secondaryValue={`$${Number(processedMetrics?.feesCollected || 0).toLocaleString()}`}
+          secondaryText="fees collected"
+          secondaryLink={DUNE_MMV2_URL}
+        />
+        <OlasIsBurnedArrow pointsDown className="mx-auto mb-2" />
+        <ActivityCard
+          icon="olas-burnt.png"
+          primaryValue={`$${Number(processedMetrics?.olasBurned || 0).toLocaleString()}`}
+          primaryText="OLAS burned"
+          primaryLink={DUNE_MMV2_URL}
+        />
+        <Image
+          src={`${imgPath}mobile-arrow5.png`}
+          alt="arrow"
+          width={287}
+          height={202}
+          className="ml-[30px]"
+        />
+      </div>
+    </div>
+  );
+};
+
+const TokenAddress = () => {
+  const [copied, setCopied] = useState(false);
+  const [activeTab, setActiveTab] = useState('Etherium');
+  const [currentAddress, setCurrentAddress] = useState(
+    '0x0001A500A6B18995B03f44bb040A5fFc28E45CB0',
+  );
+
+  const getTokenAddress = async () => {
+    try {
+      await navigator.clipboard.writeText(currentAddress);
+      setCopied(true);
+      setTimeout(() => setCopied(false), 2000);
+    } catch (error) {
+      console.error('Unable to copy to clipboard: ', error);
+    }
+  };
+
+  const [openDropdown, setOpenDropdown] = useState(false);
+
+  return (
+    <div className="place-items-center mt-20">
+      <Image
+        src="/images/homepage/olas-token.png"
+        alt="OLAS token"
+        width={487}
+        height={444}
+      />
+      <Card className="absolute card-opaque w-[90%] md:w-[648px] left-1/2 transform -translate-x-1/2 -translate-y-[200px] pt-4 p-6 bg-white flex flex-col">
+        <div className="flex md:hidden text-left w-full mb-4 relative">
+          <button
+            type="button"
+            className="w-full flex items-center justify-between bg-white rounded-lg p-2"
+            onClick={() => setOpenDropdown(!openDropdown)}
+            aria-expanded={openDropdown}
+            aria-haspopup="true"
+          >
+            <div className="flex items-center gap-2">
+              {(() => {
+                const activeItem = addresses.find(
+                  (item) => item.name === activeTab,
+                );
+                return (
+                  activeItem && (
+                    <Image
+                      src={`/images/homepage/addresses/${activeItem.activeSrc}`}
+                      alt={activeItem.name}
+                      width={20}
+                      height={20}
+                    />
+                  )
+                );
+              })()}
+              <span>{activeTab}</span>
+            </div>
+            <ChevronDown size={24} />
+          </button>
+
+          {openDropdown && (
```

Comment:
do you think it is possible to split mobileDropdown and desktop view into two components? and render like we do with `Menu` and `MenuMobile` - it will be easier to read